### DISCUSSION TITLE
slow FeatureNode performance in V3.2

### DISCUSSION THREAD

**Discussion by osgwraiESC** [pelicanmapping/osgearth#2043] [Support Q&A]
i made a ruler tool that is derived directly from the code in the terrain profile example (osgearth_terrainprofile), except in my case the second endpoint follows the mouse until you left click again,

in version 3.1 the redraw did not lag the mouse and provided a smooth user experience, in V3.2 the redraw time substantially lags the mouse move, i and i have to pause the mouse to get to see the redraw, 

any ideas as what is causing the performance hit when using AltitudeSymbol with clamping and draping from version V3.1 to V3.2? I have dug through the source and i can't find it.

i am building from source and am willing to hack some tries in there, thanks.

**Comment by nebsar** [comment]
I do not know if this helps but try to have a look at the osgearth_imgui example. Terrain GUI has a ruler that follows the mouse cursor and a label that shows the distance of the measurement. Below is the link to the source code:

https://github.com/gwaldron/osgearth/blob/master/src/osgEarth/ImGui/TerrainGUI

**Reply by osgwraiESC** [reply]
ok thanks for the heads up
